src/Math/babylon.math.ts

```diff
@@ -2227,6 +2235,27 @@
             result.m[15] = initialM44;
         }
 
+        public getRow(index: number): Vector4 {
+            if (index < 0 || index > 3) {
+                return null;
+            }
+
+            var i = index * 4;
+            return new Vector4(this.m[i + 0], this.m[i + 1], this.m[i + 2], this.m[i + 3]);
+        }
+
+        public setRow(index: number, row: Vector4): boolean {
+            if (index < 0 || index > 3) {
+                return false;
+            }
+
+            var i = index * 4;
+            this.m[i + 0] = row.x;
+            this.m[i + 1] = row.y;
+            this.m[i + 2] = row.z;
+            this.m[i + 3] = row.w;
+        }

```

Comment:
Well, I'm not a fan of fluent API for primitive like Matrix, the Invert() method is a prime example of 90% of the devs thinking it returns a new object but it doesn't.
Now for the setRow I think we can return this because the method doesn't change the whole Matrix object, so it may be easier to understand that it's 'this' that is returned.